### PR TITLE
feat(stepper): enhancements

### PR DESCRIPTION
**Pull request by adisreyaj**
## Description
Show `checked` icon for completed and non-editable steps. Gives a visual indication about the progress.

image

### CODE REVIEW

File: projects/components/src/stepper/stepper.component.ts [hunk 1]
```diff
@@ -34,17 +35,20 @@ import { StepperTabComponent } from './tab/stepper-tab.component';
...
           <mat-step
-            [completed]="step?.stepControl ? false : step.completed"
+            [completed]="step?.stepControl?.valid ?? step.completed"
             [stepControl]="step?.stepControl"
             [optional]="step.optional"
             [editable]="step.editable"
           >
             <ng-template matStepLabel>
-              <ht-label class="header-label" [label]="step.label"></ht-label>
+              <ht-label class="header-label" [attr.data-editable]="step.editable" [label]="step.label"></ht-label>
```
Comment (anandtiwary) [inline projects/components/src/stepper/stepper.component.ts:44]: why do we have to use attributes for this? Can we do it by setting a css class?

Comment (adisreyaj) [reply projects/components/src/stepper/stepper.component.ts:44]: We can.

**Comment by 23nobody** [comment]
This solution was not working the last time we tried it. Wondering what have changed since that makes it work. 😅

**Comment by 23nobody** [comment]
Please look for all the scenarios back and forth before merging.

**Comment by adisreyaj** [comment]
> e scenarios back and forth before merging.

Adding necessary icons did the job.

**Pull request merged**